assets/js/utils/rtcUtils.ts

```diff
@@ -273,17 +276,28 @@ const addOrReplaceTrack = (
   track: MediaStreamTrack,
   playerCallback: (sourceType: SourceType) => void
 ) => {
-  if (!presenterStreams[client.email]) presenterStreams[client.email] = new MediaStream();
-  const curTrack = findTrackByType(client, track.kind as SourceType);
-
-  if (curTrack && curTrack.id !== track.id && curTrack.id !== mergedScreenRef.screenTrack?.id) {
-    curTrack.stop();
+  if (presenterStreams[client.email]) {
+    const curTrack = findTrackByType(client, track.kind as SourceType);
+    if (curTrack && curTrack.id !== track.id) removeTrack(client, curTrack);
   }
+  addTrack(track, client, playerCallback);
+};
 
-  curTrack && presenterStreams[client.email].removeTrack(curTrack);
+const addTrack = (
+  track: MediaStreamTrack,
+  client: Client,
+  playerCallback: (sourceType: SourceType) => void
```

Comment:
I don't like the idea of `playerCallback`. I know that everything is based on this approach, but we could do better :)

Reply:
I couldn't agree more but AFAIC @shuntrho is changing this behaviour in [Refactor Presenter Area PR](https://github.com/membraneframework-labs/membrane_live/pull/118) so I'll leave it as it is for the moment.